src/common/snippets/src/lowered/pass/split_loops.cpp

```diff
@@ -18,23 +19,27 @@ using LoopManager = LinearIR::LoopManager;
 using LoopInfo = LoopManager::LoopInfo;
 using LoopInfoPtr = LoopManager::LoopInfoPtr;
 
-SplitLoops::SplitLoops() : Pass() {}
+SplitLoops::SplitLoops() : RangedPass() {}
 
-bool SplitLoops::can_be_split(const LoopInfoPtr& current, const LoopInfoPtr& parent) {
-    const auto current_dim_idx = current->get_dim_idx();
-    const auto parent_dim_idx = parent->get_dim_idx();
+bool SplitLoops::can_be_split(const LoopInfoPtr& loop_to_split, const LoopInfoPtr& loop_to_fuse) {
+    const auto current_dim_idx = loop_to_split->get_dim_idx();
+    const auto parent_dim_idx = loop_to_fuse->get_dim_idx();
+    const auto& handlers = loop_to_split->get_handlers();
     const bool equal_dim_idxes = current_dim_idx != LoopInfo::UNDEFINED_DIM_IDX && current_dim_idx == parent_dim_idx;
-    return current->get_work_amount() == parent->get_work_amount() && current->get_increment() != parent->get_increment() && equal_dim_idxes;
+    const bool only_main_body = handlers[LoopInfo::FIRST_ITER].empty() && handlers[LoopInfo::FIRST_ITER].empty();
```

Comment:
Could you please clarify why we have this check?

Reply:
Thanks for the finding: there was a typo in the check, I corrected it. Please let me know if the check is still unclear